src/views/Entries/QualityControl/index.tsx

```diff
@@ -1,29 +1,77 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { _cs } from '@togglecorp/fujs';
 
 import ResizableH from '#rscv/Resizable/ResizableH';
+import TableOfContents from '#components/TableOfContents';
 
 import styles from './styles.scss';
 
 interface QualityControlProps {
     className?: string;
     projectId: number;
 }
+interface Data {
+    id: number;
+    title: string;
+    children: Data[] | undefined;
+}
+
+const toc: Data[] = [
+    {
+        id: 1,
+        title: 'This is Heading',
+        children: undefined,
+    },
+    {
+        id: 2,
+        title: 'This is Second Heading',
+        children: [
+            {
+                id: 3,
+                title: 'This is inner',
+                children: undefined,
+            },
+            {
+                id: 4,
+                title: 'This is inner',
+                children: undefined,
+            },
+        ],
+    },
+];
+
+const keySelector = (d: Data) => d.id;
+const labelSelector = (d: Data) => d.title;
+const childrenSelector = (d: Data) => d.children;
 
 function QualityControl(props: QualityControlProps) {
     const {
         className,
         // projectId,
     } = props;
 
+    const [selected, setSelection] = useState<number | string | undefined>(undefined);
+
+    const handleSelection = (id: number | string) => {
+        setSelection(id);
+    };
```

Comment:
Why do we have type for `selected` as `string | undefined`
shouldn't be be either `string` or a `number`